### DISCUSSION TITLE
How to convert a set of Tube objects to gifti format

### DISCUSSION THREAD

**Discussion by willi3by** [marcomusy/vedo#1200] [General]
Hi,

I am working with vedo on a project to visualize white matter streamlines. I can successfully create a vedo plotter and visualize the streamlines with color but I am having issues exporting this information to either an obj or a gifti file. The gifti call needs vertices, faces, and colors. The vertices and colors are easy enough to extract, but when I try to access the cells of the tube, it is empty. I have also tried combining the streamlines into a single mesh using vedo.merge, but then I lose the individual streamline color information. Is there a way to extract the faces (i.e., cells) from each tube object so I can convert to gifti? Alternatively, is there a way to export as an obj while keeping the smoothness and color of the objects? Right now, if I export as obj with file_io.save and try to load in MeshLab, I get a series of vertices that are not connected and no color. 

Apologies for any naivety! I am fairly new to object visualization and have combed through the docs but haven't been able to figure this out yet.

Thanks!
Brady

**Comment by willi3by** [comment]
I figured out the issue. I didn't realize that the vertices were not already triangulated in the mesh object, so I ran the triangulate method and it generated all the faces. Also, I noticed that the obj does not output a mtl file for the coloring so I used another approach. Sorry for the premature post!

**Reply by marcomusy** [reply]
No worries! Let me know if you need help.
